File: mpi_examples/fortran_c_codes/example2/example2_mpi.f90
program mpi_example2
use mpi
implicit none
integer ierr, np, rank, message

call MPI_INIT(ierr)
call MPI_COMM_SIZE(MPI_COMM_WORLD, np, ierr)
call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)

if (rank == 0) then
    message = 48151623 

    call MPI_SEND(message, 1, MPI_INTEGER, 1, 0,&
              MPI_COMM_WORLD, ierr)

    write(*,*) "process ", rank, " sends ", message 
else if (rank == 1) then
    call MPI_RECV(message, 1, MPI_INTEGER, 0, 0,& 
              MPI_COMM_WORLD, MPI_STATUS_IGNORE, ierr)

    write(*,*) "process ",rank," receives ", message

!else if (rank == 2) then
    !write(*,*) "process ",rank," receives ", message

endif 

call MPI_Finalize(ierr) 
end program mpi_example2

File: mpi_examples/fortran_c_codes/example2/example2_mpi.c
#include <stdlib.h>
#include<stdio.h>
#include <mpi.h>

int main(int argc, char **argv){
int np, rank, message;

MPI_Init(&argc, &argv);
MPI_Comm_size(MPI_COMM_WORLD, &np);
MPI_Comm_rank(MPI_COMM_WORLD, &rank);

if (rank == 0){
    message = 48151623;

    MPI_Send(&message, 1, MPI_INT, 1, 0,
             MPI_COMM_WORLD);
    printf("Process %d sends %d\n",
           rank, message);
} 
else if(rank == 1){
    MPI_Recv(&message, 1, MPI_INT, 0, 0,
            MPI_COMM_WORLD, MPI_STATUS_IGNORE);
    printf("Process %d receives %d\n",
           rank, message);
}

MPI_Finalize();
}

